hredis/redigo/redigo.go

```diff
@@ -42,15 +44,73 @@ func WaitForAvailability(url string, d time.Duration, f WaitFunc) (bool, error)
 	}
 }
 
+// make this a var, so we can change this for testing.
+var redisDialURL = redis.DialURL
+
 // NewRedisPoolFromURL returns a new *redigo/redis.Pool configured for the supplied url
 // The url can include a password in the standard form and if so is used to AUTH against
 // the redis server
-func NewRedisPoolFromURL(url string) (*redis.Pool, error) {
+func NewRedisPoolFromURL(rawURL string, altPasses ...string) (*redis.Pool, error) {
```

Comment:
What do you think about taking option funcs instead of altPasses and making these altPasses an optionFunc? If that were done then redisDialURL could also be specified with an optionFunc and we could do away with the redisDialURL for testing purposes.

Reply:
The more I think about this, the more I'd much prefer to use option funcs for this.

Reply:
I had thought about this, but it's a bit strange as we return a `*redis.Pool`, not something we can hide state in. The redisDialURL option and the passes option both change the behavior of the `redis.Pool.Dial` function, which means we'd need to create some intermediary to run the option functions on, in order to produce an appropriate method to use for the Dial function, or a closure. I think this approach yields a lot more complexity, but I'm willing to give it a shot.